torchtitan/components/checkpoint.py

```diff
@@ -417,6 +419,7 @@ def dcp_load(
         state_dict: dict[str, Any],
         checkpoint_id: str,
         from_hf: bool,
+        dequantize: bool = False,
```

Comment:
I know this is just a draft. But I would like to discuss how we can customize Storage. I would prefer `dcp_load()` to accept a storage rather than `dequantize`. Eventually, we may want to let users to choose which storage to use.